pkg/controller/vsphere/reconciler.go

```diff
@@ -156,6 +211,10 @@ func IsNotFound(err error) bool {
 	}
 }
 
+func isRetrieveMONotFound(taskRef string, err error) bool {
+	return err.Error() == fmt.Sprintf("ServerFaultCode: The object 'vim.Task:%v' has already been deleted or has not been completely created", taskRef)
```

Comment:
This is unfortunate.

Reply:
I known, the error thrown is `soap.soapFaultError` which makes not difference I couldn't find a better way